tokio/src/time/clock.rs

```diff
@@ -208,6 +214,35 @@ cfg_test_util! {
         crate::task::yield_now().await;
     }
 
+    /// Controls time auto-advance behavior.
+    ///
+    /// When time is paused, by default the runtime automatically advances
+    /// time to the next timer. See [`pause`](pause#auto-advance) for more details.
+    ///
+    /// This function allows enabling and disabling the auto-advance behavior.
+    /// When auto-advance feature is disabled, sleeping will wait indefinitely
+    /// until it's re-enabled, or the time is manually advanced using [`advance`].
+    /// This means that if all tasks call `sleep` simultaneously, the program will
+    /// deadlock.
+    ///
+    /// # Panics
+    ///
+    /// If called from outside of the Tokio runtime.
+    ///
+    /// [`sleep`]: fn@crate::time::sleep
+    /// [`advance`]: crate::time::advance
+    pub fn set_time_auto_advance(enable: bool) {
```

Comment:
Given that the fn is in the `time` module, including `time` in the fn name is redundant: `time::set_time_auto_advance(true)`.

Also, does this function need to exist (I couldn't say because I don't fully understand the use case yet).